drivers/include/periph/rtc.h

```diff
@@ -208,6 +208,30 @@ void rtc_localtime(uint32_t time, struct tm *t);
  */
 bool rtc_tm_valid(const struct tm *t);
 
+/**
+ * @brief Set the time as epoch with sub-second precision
+ *        This feature is an extension provided by the `rtt_rtc` module.
+ *
+ * @note The actual µs precision depends on the underlying hardware.
+ *       The smallest time step will be 1 / @ref RTT_FREQUENCY.
+ *
+ * @param[in] s     The new epoch timestamp
+ * @param[in] us    Sub-Seconds
+ */
+void rtc_settimeofday(uint32_t s, uint32_t us);
+
+/**
+ * @brief Get the current epoch with sub-second precision
+ *        This feature is an extension provided by the `rtt_rtc` module.
+ *
+ * @note The actual µs precision depends on the underlying hardware.
+ *       The smallest time step will be 1 / @ref RTT_FREQUENCY.
+ *
+ * @param[out] s    The current epoch timestamp
+ * @param[out] us   Sub-Seconds
+ */
+void rtc_gettimeofday(uint32_t *s, uint32_t *us);
```

Comment:
I don't like implementation specific APIs being exposed in the common header. IMO the following two alternatives should be taken:

1. Make this a feature and no longer specific to the `rtc_rtt` implementation
    - But does this API really make sense for hardware implementations? More likely a this could be a kind of convenience function on top of `rtc_set_time()` 
2. Move this to an `rtc_rtt.h` header

Reply:
I think the second solution is the cleanest.
I really can't think of any *real* RTC that lets you *set* the sub-second component. 

There are some that expose the internal counter for reading, but won't let you mess with it.

If those show up in the future we can always move those functions back to `periph/rtc.h`.